torch/nn/modules/distance.py

```diff
@@ -7,13 +7,26 @@
 
 class PairwiseDistance(Module):
     r"""
-    Computes the pairwise distance between vectors :math:`v_1`, :math:`v_2` using the p-norm:
+    Computes the pairwise distance between input vectors, or between columns of input matrices.
+
+    Distances are computed using ``p``-norm, with constant ``eps`` added to avoid division by zero
+    if ``p`` is negative, i.e.:
+
+    .. math ::
+        \mathrm{dist}\left(\mathbf{v}, \mathbf{u}\right) = \left\Vert \mathbf{v}-\mathbf{u} + \epsilon \mathbf{e} \right\Vert_p,
+
+    where :math:`\mathbf{e}` is the vector of ones and the ``p``-norm is given by.
 
     .. math ::
-        \Vert x \Vert _p = \left( \sum_{i=1}^n  \vert x_i \vert ^ p \right) ^ {1/p}.
+        \Vert \mathbf{x} \Vert _p = \left( \sum_{i=1}^n  \vert x_i \vert ^ p \right) ^ {1/p}.
+
+    Returns either a scalar (if input tensors are rank 1) or a vector of a vector of all pairs of distances
```

Comment:
for tensors of rank 1 or 0 it returns 0d tensor, not scalar. Also, what is `vector of a vector`?  It's a 1d tensor with distances. It also works for tensors with a rank higher than 2

Reply:
You are right, I'll fix that. I was thinking of `pdist` when I wrote that, but this is for `pairwise_distance`.